cmd/clusterctl/pkg/client/config/provider.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+)
+
+// Provider defines a provider configuration.
+type Provider interface {
+	Name() string
+	URL() string
+	Type() clusterctlv1.ProviderType
+}
+
+// provider implements provider
+type provider struct {
+	name  string
+	url   string
+	ttype clusterctlv1.ProviderType
```

Comment:
Would it be better to just call this `providerType` instead of `ttype` here? It isn't clear what `ttype` means other than a way to work around `type` being a golang builtin.

Reply:
fixed!